usersync/ejector.go

```diff
@@ -0,0 +1,127 @@
+package usersync
+
+import (
+	"errors"
+	"time"
+)
+
+type Ejector interface {
+	Choose(uids map[string]UIDEntry) (string, error)
+}
+
+type OldestEjector struct{}
+
+type PriorityBidderEjector struct {
+	PriorityGroups   [][]string
+	syncersByBidder  map[string]Syncer
+	IsSyncerPriority bool
+	TieEjector       Ejector
+}
+
+// Choose method for oldest ejector will return the oldest uid
+func (o *OldestEjector) Choose(uids map[string]UIDEntry) (string, error) {
+	var oldestElement string
+	var oldestDate time.Time = time.Unix(1<<63-62135596801, 999999999) // Max value for time
+
+	for key, value := range uids {
+		if value.Expires.Before(oldestDate) {
```

Comment:
Nitpick: This unlikely edge case where `cookie.uids` elements come with the max possible value for the `time.Time` type would make the `for` loop in line 61 of `usersync/cookie.go` an infinite loop. Should we add an extra `if len(uidToDelete) > 0` check after the `ejector.Choose(cookie.uids)` call in line 83 of `usersync/cookie.go`?
```
cookie.uids := map[string]UIDEntry{
    "maxExpTimeEntry": { 
        UID: "123",
        Expires: time.Unix(1<<63-62135596801, 999999999),
    },
    "anotherMaxExpTimeEntry": { 
        UID: "456",
        Expires: time.Unix(1<<63-62135596801, 999999999),
    },
}
priorityGroups := [][]string{}
```
Or maybe modify the `OldestEjector` to return a random element of `uids` if `oldestElement` is still empty in line 32 of `usersync/ejector.go`? We used to do that in a previous delta, didn't we?

These two extra test cases in `TestPriorityEjector` would also make `test.givenEjector.Choose(test.givenUids)` return an empty string
```
    {
        name: "max-expiry-time",
        givenUids: map[string]UIDEntry{
            "lowestPriority": {
                UID:     "456",
                Expires: time.Unix(1<<63-62135596801, 999999999),
            },
        },
        givenEjector: &PriorityBidderEjector{
            PriorityGroups: [][]string{
                {"highestPriority2", "highestPriority3"},
                {"lowestPriority2", "lowestPriority"},
            },
            syncersByBidder: map[string]Syncer{
                "highestPriorityBidder": fakeSyncer{
                    key: "highestPrioritySyncer",
                },
                "lowestPriority": fakeSyncer{
                    key: "lowestPriority",
                },
            },
            IsSyncerPriority: true,
            TieEjector:       &OldestEjector{},
        },
        expected: "lowestPriority",
    },
    {
        name: "empty-priority-group-matrix",
        givenUids: map[string]UIDEntry{
            "lowestPriority": {
                UID:     "456",
                Expires: time.Unix(1<<63-62135596801, 999999999),
            },
        },
        givenEjector: &PriorityBidderEjector{
            PriorityGroups: [][]string{},
            syncersByBidder: map[string]Syncer{
                "highestPriorityBidder": fakeSyncer{
                    key: "highestPrioritySyncer",
                },
                "lowestPriority": fakeSyncer{
                    key: "lowestPriority",
                },
            },
            IsSyncerPriority: true,
            TieEjector:       &OldestEjector{},
        },
        expected: "lowestPriority",
    },
```